src/babashka/curl.clj

```diff
@@ -174,23 +175,39 @@
                   :headers headers
                   :body body
                   :err err
-                  :process (:proc opts)}]
+                  :process process
+                  :exit exit}]
     response))
 
 ;;;; End Response Parsing
 
+(def unexceptional-status?
+  #{200 201 202 203 204 205 206 207 300 301 302 303 304 307})
+
+(defn- should-throw? [response opts]
+  (let [exceptional-status? (not (unexceptional-status? (:status response)))
+        nonzero-exit? (not (zero? (:exit response)))]
+    (and (:throw-exceptions opts)
+         (or exceptional-status? nonzero-exit?))))
+
 (defn request [opts]
   (let [header-file (File/createTempFile "babashka.curl" ".headers")
         opts (assoc opts :header-file header-file)
+        default-opts {:throw-exceptions true}
```

Comment:
Using the keyword `:throw-exceptions` to match [clj-http's keyword.](https://github.com/dakrone/clj-http/blob/3.x/src/clj_http/client.clj#L235)

Reply:
Since we already omit the question mark, what about just `:throw`?

Reply:
Sounds good. I'll change it to just `:throw`. While we're on the subject, what's your stance on how close this API should be to clj-http's API?

Reply:
In the README I wrote that this library is inspired by clj-http. But I don't think we should have to mimick everything to the letter.